### DISCUSSION TITLE
wifi features

### DISCUSSION THREAD

**Discussion by tarxvftech** [tarxvftech/FCS-152#40] [Ideas]
What cool things would be _easy_ with the wifi feature but interesting enough to draw people in (and maybe earn some goodwill with FCS for future collaborations?)

* A simple web interface for codeplug editing?
* Easy codeplug sharing?
* Firmware updates right off of github releases?

* [ ] PoC: Flash firmware from github releases or equivalent (can I use a url shortener? branches+build artifacts?)
* [x] Wifi support that's reasonable - not hardcoded.

**Comment by AlexGlide** [comment]
First of all I think firmware updates will be very necessary. After that I prefer a easy codeplug sharing.

**Comment by tarxvftech** [comment]
Firmware updates for sure, then. That also implies it has to be configurable for STA mode. Easiest way to do that, considering the limited options for input, is probably to configured that over the web interface from a default AP mode. 
I know there's a lot of code out there that makes that easy, courtesy of arduino.

**Comment by tarxvftech** [comment]
Wifi support added, including softAP with captive portal. Also includes a local http firmware upgrade over wifi.
With great thanks to the author(s) of [AutoConnect](https://hieromon.github.io/AutoConnect/index.html)!

**Comment by tarxvftech** [comment]
I'd like to have a menu button to auto-flash the latest release from a particular branch on github. That's a 'later' type of problem I think, as the radio itself is currently unusable and the menu system needs a bunch of attention before we can get there. But the wifi works and a much easier local flash method is in place for development, no serial cable needed!
